python/tvm/driver/tvmc/compiler.py

```diff
@@ -191,22 +198,21 @@ def compile_model(
 
         if use_autoscheduler:
             with auto_scheduler.ApplyHistoryBest(tuning_records):
-                with tvm.transform.PassContext(
-                    opt_level=3, config={"relay.backend.use_auto_scheduler": True}
-                ):
+                config["relay.backend.use_auto_scheduler"] = True
+                with tvm.transform.PassContext(opt_level=3, config=config):
                     logger.debug("building relay graph with autoscheduler")
                     graph_module = relay.build(
                         mod, target=target, params=params, target_host=target_host
                     )
         else:
             with autotvm.apply_history_best(tuning_records):
-                with tvm.transform.PassContext(opt_level=3):
+                with tvm.transform.PassContext(opt_level=3, config=config):
                     logger.debug("building relay graph with tuning records")
                     graph_module = relay.build(
                         mod, tvm_target, params=params, target_host=target_host
                     )
     else:
-        with tvm.transform.PassContext(opt_level=3):
+        with tvm.transform.PassContext(opt_level=3, config=config):
```

Comment:
[Suggestion] I think the setting the config should be tested for additional arguments of byoc targets.

Reply:
The default value for `config` within `PassContext` is `None`. So in this case I'd be able to replace `config=config` by `config=config or None`, but it would be redundant with a check already done in the `PassContext` constructor, which reads (keeping only the relevant bits):

```
def __init__(
    self, opt_level=2, required_pass=None, disabled_pass=None, trace=None, config=None
):
    ...
    config = config if config else None
    self.__init_handle_by_constructor__(
        _ffi_transform_api.PassContext, opt_level, required, disabled, trace, config
    )
    ...
```

Based on that, I'm inclined to keep it as proposed here, but happy to discuss.

Reply:
Sorry if I was not clear before :) 
Well I was referring to opts that is parsed from the CLI and not questioning the ablity of PassContext accepting the config options.

I think its within the scope of this patch to introduce CLI parsing of additional opts that goes as config of the PassContext. So we might want to make sure what the user gives as opts to the CLI ends up in the PassContext correctly.

Reply:
Sure, will add a test case for that.

Reply:
In the last version of this PR, see `test_compile_check_configs_byoc`